public/app/features/chat/Chat.tsx

```diff
@@ -0,0 +1,293 @@
+    if (c) {
+      this.subscription = c.subscribe({
+        next: (msg) => {
+          if (isLiveChannelMessageEvent(msg)) {
+            const { messageCreated } = msg.message;
+            if (messageCreated) {
+              this.setState((prevState) => ({
+                messages: [...prevState.messages, messageCreated],
+              }));
+              this.scrollToBottom();
+            }
+          }
+        },
+      });
+    }
+  };
+
+  render() {
+    let messageElements;
+    if (this.state.messages.length > 0) {
+      messageElements = (
+        <div className="chat-messages">
+          {this.state.messages.map((msg) => (
+            <ChatMessage key={msg.id} message={msg} actions={this.props.actions} />
+          ))}
+        </div>
+      );
+    } else {
+      messageElements = <div className="chat-messages">No messages here yet</div>;
+    }
+
+    return (
+      <div
+        className="chat"
+        ref={(el) => {
+          this.chatContainer = el;
+        }}
+      >
+        {messageElements}
+        <TextArea
+          placeholder="Write a message"
+          value={this.state.value}
+          onChange={this.handleChange}
+          onKeyUp={this.onKeyboardUp}
+          onKeyPress={this.onKeyboardPress}
+          ref={(el) => {
+            this.chatInput = el;
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+interface ChatMessageAction {
+  verbal: string;
+  action: any;
+}
+
+interface ChatMessageProps {
+  message: Message;
+  actions?: ChatMessageAction[];
+}
+
+const ChatMessage: FunctionComponent<ChatMessageProps> = ({ message, actions = [] }) => {
+  let senderColor = '#34BA18';
+  let senderName = 'System';
+  let avatarUrl = '/public/img/grafana_icon.svg';
+  if (message.userId > 0) {
+    senderColor = '#19a2e7';
+    senderName = message.user.login;
+    avatarUrl = message.user.avatarUrl;
+  }
+  const timeColor = '#898989';
+  const timeFormatted = new Date(message.created * 1000).toLocaleTimeString();
+  const markdownContent = renderChatMarkdown(message.content);
+
+  // const [actionMenuExpanded, setActionMenuExpanded] = useState(false);
+  const [showActionIcon, setShowActionIcon] = useState(false);
+
+  const renderMenuGroupItems = () => {
+    return (
+      <MenuGroup label="">
+        {actions.map((action) => (
+          <MenuItem
+            key={action.verbal}
+            label={action.verbal}
+            onClick={() => {
+              action.action(message);
+            }}
+          />
+        ))}
+      </MenuGroup>
+    );
+  };
+  return (
+    <div
+      onMouseEnter={() => {
+        setShowActionIcon(true);
+      }}
+      onMouseLeave={() => {
+        setShowActionIcon(false);
+      }}
+      className="chat-message"
+    >
+      <div style={{ float: 'left', paddingTop: '6px', marginRight: '10px' }}>
+        <img src={avatarUrl} alt="" style={{ width: '30px', height: '30px' }} />
+      </div>
+      <div style={{ position: 'relative' }}>
```

Comment:
Can use [docs here](https://github.com/grafana/grafana/blob/main/contribute/style-guides/styling.md) and create a `getStyles` func that returns classes vs. doing inline styling here

Inline styling for colors is fine given they are dynamic